src/main/java/com/example/pokebattlez/battle/model/Pokemon.java

```diff
@@ -0,0 +1,83 @@
+package com.example.pokebattlez.battle.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import static com.example.pokebattlez.battle.model.Stat.*;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class Pokemon {
+    private static final float NATURE_MODIFIER = 0.1f;
+
+    private String name;
+    private int position;
+
+    private int level;
+
+    private Map<com.example.pokebattlez.battle.model.Stat, Stat> stats = Map.of(
+            HP, new Stat(Stat::calculateHp),
+            ATTACK, new Stat(),
+            DEFENCE, new Stat(),
+            SP_ATTACK, new Stat(),
+            SP_DEFENCE, new Stat(),
+            SPEED, new Stat()
+    );
```

Comment:
Having Stat enum and Stat class (inside the Pokemon) is complicating the code comprehension.  
I guess Stat enum means more StatType...

Reply:
This should be the only place where the Stat class gets used explicitly. But I am gonna refactor the name of it to maybe StatData.